src/states/infiniteweightpeps.jl

```diff
@@ -0,0 +1,211 @@
+    return 2 * prod(size(wts.x))
+end
+
+function Base.isapprox(wts1::SUWeight, wts2::SUWeight; atol=0.0, rtol=1e-5)
+    return (
+        isapprox(wts1.x, wts2.x; atol=atol, rtol=rtol) &&
+        isapprox(wts1.y, wts2.y; atol=atol, rtol=rtol)
+    )
+end
+
+function compare_weights(wts1::SUWeight, wts2::SUWeight)
+    wtdiff = sum(_singular_value_distance((wt1, wt2)) for (wt1, wt2) in zip(wts1, wts2))
+    return wtdiff / (2 * prod(size(wts1)))
+end
+
+"""
+Represents an infinite projected entangled-pair state on a 2D square lattice
+consisting of vertex tensors and bond weights
+"""
+struct InfiniteWeightPEPS{T<:PEPSTensor,E<:PEPSWeight} <: AbstractPEPS
+    vertices::Matrix{T}
+    weights::SUWeight{E}
+
+    function InfiniteWeightPEPS(
+        vertices::Matrix{T}, weights::SUWeight{E}
+    ) where {T<:PEPSTensor,E<:PEPSWeight}
+        @assert size(vertices) == size(weights)
+        Nr, Nc = size(vertices)
+        for (r, c) in Iterators.product(1:Nr, 1:Nc)
+            space(weights.y[r, c], 1)' == space(vertices[r, c], 2) || throw(
+                SpaceMismatch("South space of bond weight y$((r, c)) does not match.")
+            )
+            space(weights.y[r, c], 2)' == space(vertices[_prev(r, Nr), c], 4) || throw(
+                SpaceMismatch("North space of bond weight y$((r, c)) does not match.")
+            )
+            space(weights.x[r, c], 1)' == space(vertices[r, c], 3) ||
+                throw(SpaceMismatch("West space of bond weight x$((r, c)) does not match."))
+            space(weights.x[r, c], 2)' == space(vertices[r, _next(c, Nc)], 5) ||
+                throw(SpaceMismatch("West space of bond weight x$((r, c)) does not match."))
+        end
+        return new{T,E}(vertices, weights)
+    end
+end
+
+"""
+Create an InfiniteWeightPEPS from matrices of vertex tensors,
+x-weights and y-weights
+"""
+function InfiniteWeightPEPS(
+    vertices::Matrix{T}, wts_x::Matrix{E}, wts_y::Matrix{E}
+) where {T<:PEPSTensor,E<:PEPSWeight}
+    return InfiniteWeightPEPS(vertices, SUWeight(wts_x, wts_y))
+end
+
+"""
+Create an InfiniteWeightPEPS by specifying its physical, north and east spaces and unit cell.
+Spaces can be specified either via `Int` or via `ElementarySpace`.
+Bond weights are initialized as identity matrices. 
+"""
+function InfiniteWeightPEPS(
+    f, T, Pspace::S, Nspace::S, Espace::S=Nspace; unitcell::Tuple{Int,Int}=(1, 1)
+) where {S<:ElementarySpace}
+    vertices = InfinitePEPS(f, T, Pspace, Nspace, Espace; unitcell=unitcell).A
+    weights = SUWeight(fill(id(Espace), unitcell), fill(id(Nspace), unitcell))
+    return InfiniteWeightPEPS(vertices, weights)
+end
+
+"""
+Absorb environment weight on axis `ax` into tensor `t` at position `(row,col)`
+
+Weights around the tensor at `(row, col)` are
+```
+                ↓
+                y[r,c]
+                ↓
+    ←x[r,c-1] ← T[r,c] ← x[r,c] ←
+                ↓
+                y[r+1,c]
+                ↓
+```
+"""
+function absorb_wt(
```

Comment:
```suggestion
function absorb_weight(
```

I find that `wt` is not very clear as an abbreviation, especially in function names and kwargs. Would you consider using `weight` instead? (I care less about this for internal variable names, although typically it is slightly more readable to simply write out names)